pkg/controller/mizar/mizar-pod-controller.go

```diff
@@ -121,13 +121,18 @@ func (c *MizarPodController) updateObj(old, cur interface{}) {
 	}
 
 	key, _ := controller.KeyFunc(curObj)
-	c.queue.Add(KeyWithEventType{Key: key, EventType: EventType_Update})
+	c.queue.Add(KeyWithEventType{Key: key, EventType: EventType_Update, ResourceVersion: curObj.ResourceVersion})
 }
 
 func (c *MizarPodController) deleteObj(obj interface{}) {
 	key, _ := controller.KeyFunc(obj)
 	klog.Infof("%v deleted. key %s.", controllerForMizarPod, key)
-	c.queue.Add(KeyWithEventType{Key: key, EventType: EventType_Delete})
+	c.queue.Add(KeyWithEventType{Key: key, EventType: EventType_Delete, ResourceVersion: getResourceVersion(obj)})
+}
+
+func getResourceVersion(obj interface{}) string {
+	pod := obj.(*v1.Pod)
```

Comment:
Let's re-use the apimachinery util func to do the job:
```
import "k8s.io/apimachinery/pkg/api/meta"

func GetResourceVersion(obj interface{}) string {
     objAccessor, _ := meta.Accessor(obj)
    return objAccessor.GetResourceVersion()
}
```

Reply:
Updated. Thanks for writing the code.